### PR TITLE
[Compute] `az capacity reservation group create/update`: Add new parameter `--sharing-profile` to support sharing capacity reservation group across subscriptions

### PR DESCRIPTION
**Pull request by Jing-song**
**Related command**

az capacity reservation group create/update

**Description**

Changes related to sharing a capacity reservation group. https://github.com/Azure/azure-cli/issues/28327

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Compute

File: src/azure-cli/azure/cli/command_modules/vm/_params.py [hunk 1]
```diff
@@ -1581,6 +1581,7 @@ def load_arguments(self, _):
         c.argument('capacity_reservation_group_name', options_list=['--capacity-reservation-group', '-n'],
                    help='The name of the capacity reservation group.')
         c.argument('tags', tags_type)
+        c.argument('sharing_profile', nargs='+', help='Space-separated subscription resource IDs.Specify the settings to enable sharing across subscriptions for the capacity reservation group resource.')
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/vm/_params.py:1584]:
```suggestion
        c.argument('sharing_profile', nargs='+', help='Space-separated subscription resource IDs. Specify the settings to enable sharing across subscriptions for the capacity reservation group resource.')
```

File: src/azure-cli/azure/cli/command_modules/vm/tests/latest/test_vm_commands.py [hunk 2]
```diff
@@ -9937,6 +9937,19 @@ def test_capacity_reservation_list_delete(self, resource_group):
             time.sleep(60)
         self.cmd('capacity reservation group delete -n {reservation_group} -g {rg} --yes')
 
+    @record_only()
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/vm/tests/latest/test_vm_commands.py:9940]: Please add comments in the code to explain why the test should be marked as ` @record_only()`

File: src/azure-cli/azure/cli/command_modules/vm/tests/latest/test_vm_commands.py [hunk 3]
```diff
@@ -9937,6 +9937,19 @@ def test_capacity_reservation_list_delete(self, resource_group):
...
+    def test_capacity_reservation_sharing_profile(self, resource_group):
+
+        self.kwargs.update({
+            'reservation_group_1': self.create_random_name('reservation_group_', 30),
+            'reservation_group_2': self.create_random_name('reservation_group_', 30),
+            'sub_id': '/subscriptions/7a624c46-eaa3-4a6c-b362-77c14bf531be'
+        })
+        self.cmd('capacity reservation group create -n {reservation_group_1} -g {rg} --sharing-profile "subscriptions/7a624c46-eaa3-4a6c-b362-77c14bf531be" -l westEurope')
+        self.cmd('capacity reservation group create -n {reservation_group_2} -g {rg} -l westEurope')
+        self.cmd('capacity reservation group create -n {reservation_group_2} -g {rg} --sharing-profile "subscriptions/7a624c46-eaa3-4a6c-b362-77c14bf531be" -l westEurope')
```
Comment (yanzhudd) [inline src/azure-cli/azure/cli/command_modules/vm/tests/latest/test_vm_commands.py:9951]: Should this be `update` command?
```suggestion
        self.cmd('capacity reservation group update -n {reservation_group_2} -g {rg} --sharing-profile "subscriptions/7a624c46-eaa3-4a6c-b362-77c14bf531be" -l westEurope')
```

**Review by yanzhudd — APPROVED** [review]

**Pull request merged**